oof ouch my bones

actioninja opened the pull request:
:cl: actioninja
add: Bone Hurting Juice
/:cl:

[why]: ~~I've had the idea for a while and got daybanned so decided to finally do it.  It's a (pretty) weak toxin that does light brute damage pretty quickly.  1 brute per 2 units.  If overdosed there's a very low chance every tick for it to remove a limb, but the overdose threshold is 60 and it processes 2 units every tick.  Open to any balance suggestions.~~

~~That was a really terrible idea so it does stamina damage now only.  Same rate , 1:2, overdose was lowered to 50 units.  Now instead of ripping limbs off it causes a massive amount of stamina damage to that one limb to disable it for a while.~~

~~Recipe is 1 part unstable mutagen to 3 parts itching powder to 1 part milk.  It doesn't make much sense but I figured itching powder isn't SUPER easy to make and the mutagen is "mutating" the itching powder into something else.  Also MILK.  I'm open to better recipe suggestions.~~

I now have a more finalized version of it.
The recipe is 3 itching powder, 1 unstable mutagen, 1 milk
On normal usage, it does 15 stamina damage a cycle and consumes at a rate of 2 units per cycle.  If you're a skellyton or a plasmaman it does 0.5 brute as well.
On overdose, it has a small chance of causing a huge spike of stamina damage to a random limb, temporarily disabling it.  If you're a skellyton or a plasmaman, it does the same thing but with brute instead of stamina.
There's some other effects as well such as some forced speech but those are all not really gameplay affecting.

actioninja committed:
oof ouch

XDTM commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,62 @@
+	name = "Bone Hurting Juice"
+	id = "bonehurtingjuice"
+	description = "A strange substance that looks a lot like water. Drinking it is oddly tempting. Oof ouch."
+	color = "#AAAAAA77"
```
AAAAAAAA

actioninja replied:
AAAAAA is actually just the color for water but I thought it was appropriate.

Cyberboss replied:
This isn't a proper hex colour?

actioninja replied:
It is, the 4th byte is transparency.

XDTM commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,62 @@
+		var/list/possibleSounds = list('sound/misc/desceration-01.ogg','sound/misc/desceration-02.ogg','sound/misc/desceration-01.ogg')
+		var/extMessage = "<span class='warning'>[M]'s bones hurt too much!!</span>"  //I couldn't figure out how to define a new function so let's not repeat the same strings 4 times.
+		var/playerMessage = "<span class='danger'>Your bones hurt too much!!</span>"
+		switch(pick(1, 2, 3, 4)) //Yeah, I know that this makes it so that the limb removal can fail if the limb is already gone, it doesn't pick another limb.
```
you might want to consider dealing a spike of stamina damage to a single limb, disabling it temporarily

it would make more sense, since it's bone hurting more than bone disintegrating

XDTM commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,62 @@
+	M.say("oof ouch my bones")
+
+/datum/reagent/toxin/bonehurtingjuice/on_mob_life(mob/living/carbon/M)
+	M.adjustBruteLoss(1.0, 0)
```
i suggest stamina over brute, it hurts but not necessarily harms

actioninja replied:
I forgot stamina damage was a thing while making this and that's much, much better of an idea.

actioninja committed:
Heavy nerf because it did way more than I thought it did in relation to other chems.  I mean, oof ouch.

actioninja committed:
massive rework to be more obnoxious than crippling.  I don't know what I was thinking with the first pass.

actioninja committed:
nah thats too much stamdam

actioninja committed:
small balance change

actioninja committed:
milc

XDTM commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,68 @@
+	M.say("oof ouch my bones")
+
+/datum/reagent/toxin/bonehurtingjuice/on_mob_life(mob/living/carbon/M)
+	M.adjustStaminaLoss(1.0, 0)
```
Should be higher, remember people regenerate stamina damage rapidly

GuyonBroadway commented:
Stealing yet more hippie memes eh?


https://github.com/HippieStation/HippieStation/pull/2484


Jokes aside you should totally make it do more damage to plasmamen/skeletons cus they are all bone.

actioninja commented:
I didn't know it was even a thing on hippie, I just thought this game needs more dead memes.  I'll look into plasmeme and skele bonus damage because that's hilarious.

actioninja committed:
huge stamina damage buff

Cyberboss commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,68 @@
+	M.say("oof ouch my bones")
+
+/datum/reagent/toxin/bonehurtingjuice/on_mob_life(mob/living/carbon/M)
+	M.adjustStaminaLoss(8.0, 0)
```
Remove decimal

actioninja replied:
done

Cyberboss commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,68 @@
+	var/obj/item/bodypart/r_leg = M.get_bodypart(BODY_ZONE_R_LEG)
+	. = ..()
+	if(prob(4) && iscarbon(M) && (r_arm || l_arm || l_leg || r_leg)) //big oof
+		var/list/possibleSounds = list('sound/misc/desceration-01.ogg','sound/misc/desceration-02.ogg','sound/misc/desceration-01.ogg')
```
use the get_sfx proc

actioninja replied:
done

Carbonhell commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,68 @@
+	metabolization_rate = 2.0
+
+/datum/reagent/toxin/bonehurtingjuice/on_mob_add(mob/living/carbon/M)
+	M.say("oof ouch my bones")
```
might be wise to add the forced argument to specify this was forced by the chemical
( https://github.com/tgstation/tgstation/pull/39964 )

actioninja replied:
done

Carbonhell commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,68 @@
+		switch(pick(1, 2, 3))
+			if(1)
+				var/list/possibleSays = list("oof", "ouch", "my bones", "oof ouch", "oof ouch my bones")
+				M.say(pick(possibleSays))
```
same

Carbonhell commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,68 @@
+				M.say(pick(possibleSays))
+			if(2)
+				var/list/possibleMes = list("oofs softly", "looks like their bones hurt", "grimaces, as though their bones hurt")
+				M.say("*custom " + pick(possibleMes))
```
same

Carbonhell commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,68 @@
+					l_arm.receive_damage(0, 0, stamDamage)
+					playsound(M,pick(possibleSounds), 50, TRUE, -1)
+					M.visible_message(extMessage, playerMessage)
+					M.say(ouchie)
```
same

Carbonhell commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,68 @@
+					r_arm.receive_damage(0, 0, stamDamage)
+					playsound(M,pick(possibleSounds), 50, TRUE, -1)
+					M.visible_message(extMessage, playerMessage)
+					M.say(ouchie)
```
same

Cyberboss commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,68 @@
+				M.say("*custom " + pick(possibleMes))
+			if(3)
+				to_chat(M, "<span class='warning'>Your bones hurt!</span>")
+	. = ..()
```
return ..()

Carbonhell commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,68 @@
+					l_leg.receive_damage(0, 0, stamDamage)
+					playsound(M,pick(possibleSounds), 50, TRUE, -1)
+					M.visible_message(extMessage, playerMessage)
+					M.say(ouchie)
```
same

Carbonhell commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,68 @@
+					r_leg.receive_damage(0, 0, stamDamage)
+					playsound(M,pick(possibleSounds), 50, TRUE, -1)
+					M.visible_message(extMessage, playerMessage)
+					M.say(ouchie)
```
same

Cyberboss commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,68 @@
+	toxpwr = 0
+	taste_description = "bone hurting"
+	overdose_threshold = 50
+	metabolization_rate = 2.0
```
Remove decimal

actioninja replied:
done

Cyberboss commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,68 @@
+					l_arm.receive_damage(0, 0, stamDamage)
+					playsound(M,pick(possibleSounds), 50, TRUE, -1)
+					M.visible_message(extMessage, playerMessage)
+					M.say(ouchie)
```
A lot of duplicated code, just take the last 3 line of every case of the switch

actioninja replied:
yeah that was garbage, it's been resolved now.

Cyberboss commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,68 @@
+		var/playerMessage = "<span class='danger'>Your bones hurt too much!!</span>"
+		var/ouchie = "OOF!!"
+		var/stamDamage = 200
+		switch(pick(1, 2, 3, 4)) //God help you if the same limb gets picked twice quickly.
```
use rand(1,  4)

actioninja replied:
done

Cyberboss commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,68 @@
+		var/extMessage = "<span class='warning'>[M]'s bones hurt too much!!</span>"  //I couldn't figure out how to define a new function so let's not repeat the same strings 4 times.
+		var/playerMessage = "<span class='danger'>Your bones hurt too much!!</span>"
+		var/ouchie = "OOF!!"
+		var/stamDamage = 200
```
None of these 4 need to be vars

actioninja replied:
stam damage will stay as a var to make changing the limb damage possible with just one variable.  The other three were only vars because of really bad coding so they've been dropped.

Cyberboss commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,68 @@
+	var/obj/item/bodypart/l_arm = M.get_bodypart(BODY_ZONE_L_ARM)
+	var/obj/item/bodypart/r_arm = M.get_bodypart(BODY_ZONE_R_ARM)
+	var/obj/item/bodypart/l_leg = M.get_bodypart(BODY_ZONE_L_LEG)
+	var/obj/item/bodypart/r_leg = M.get_bodypart(BODY_ZONE_R_LEG)
```
You should only need one var

actioninja replied:
I'm not really sure what you mean by this, but I just dropped all the vars entirely in favor of direct references.  Does it error if you attempt to damage a missing limb though?

Cyberboss commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,68 @@
+		switch(pick(1, 2, 3, 4)) //God help you if the same limb gets picked twice quickly.
+			if(1)
+				if(l_arm)
+					l_arm.receive_damage(0, 0, stamDamage)
```
Which is assigned in the cases

actioninja replied:
not sure what you mean by this

Cyberboss reviewed (changes requested):
All the say's should have `forced` set to `bonehurtingjuice`

actioninja committed:
big ol code cleanup

actioninja commented:
The code should suck a lot less dick now since I learned a bit more about DM.
Working on plasmeme/skellyton bonus effects now.

actioninja committed:
I should compile before pushing

XDTM commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -889,63 +889,42 @@
-					playsound(M,pick(possibleSounds), 50, TRUE, -1)
-					M.visible_message(extMessage, playerMessage)
-					M.say(ouchie)
+				M.get_bodypart(BODY_ZONE_L_ARM).receive_damage(0, 0, stamina_damage)
```
This will runtime if the mob has no right arm; you need to check that it exists before doing anything with it. Same goes for the other limbs below.

XDTM commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -889,63 +889,42 @@
-	var/obj/item/bodypart/l_leg = M.get_bodypart(BODY_ZONE_L_LEG)
-	var/obj/item/bodypart/r_leg = M.get_bodypart(BODY_ZONE_R_LEG)
-	. = ..()
 	if(prob(4) && iscarbon(M) && (r_arm || l_arm || l_leg || r_leg)) //big oof
```
remove the limb check here, put it in the switch below

Cyberboss commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,47 @@
+			if(3)
+				M.get_bodypart(BODY_ZONE_L_LEG).receive_damage(0, 0, stamina_damage)
+			if(4)
+				M.get_bodypart(BODY_ZONE_R_LEG).receive_damage(0, 0, stamina_damage)
```
Don't use `.` after procs. It's equivalent to using `:`

Cyberboss replied:
What you should do is assign the bodypart enum to a var in the switch and then do 
```dm
var/obj/item/bodypart/bp = M.get_bodypart(selected_part)
bp.receive_damage(0, 0, stamina_damage)
```

actioninja replied:
I see what you mean now, that also allows you to roll the 4 separate if checks into one.

actioninja replied:
This has been fixed now

actioninja committed:
more cleanup, runtime error fix, bonus memes

actioninja commented:
Ok now it doesn't even compile again, don't review again until after I fix that.

actioninja committed:
plz compile

actioninja committed:
THIS TIME IT SHOULD WORK

I'm not at home so am just pushing and seeing if travis compiles it at this point

Lxtoofonea commented:
+1 lel

actioninja committed:
fuck fucking dm

actioninja committed:
bonus skellington effects

actioninja committed:
stamina damage does even less than I though

actioninja commented:
I think this is pretty much ready now.  I could potentially improve the overdose code but I don't know if it would really be worth it.

XDTM commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,65 @@
+			if(4)
+				selected_part = BODY_ZONE_R_LEG
+		var/obj/item/bodypart/bp = M.get_bodypart(selected_part)
+		if(M.dna.species.type == /datum/species/skeleton || M.dna.species.type == /datum/species/plasmaman) //We're so sorry skeletons, you're so misunderstood
```
This should be inverted, you put the non-skeleton code under here and the skeleton code for non-skeletons

actioninja replied:
wew lad I did

XDTM commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,65 @@
+				bp.receive_damage(200, 0, 0) //But I don't think we should
+			else
+				to_chat(M, "<span class='warning'>Your missing arm aches from whever you left it.</span>")
+				M.say("*custom sighs", forced="bonehurtingjuice")
```
M.emote("sigh")

actioninja replied:
does emote have the forced parameter?  If it doesn't I'm going to keep it like that.

XDTM replied:
I doubt anyone's going to check logs about a sigh, and we have tons of forced emotes already.

epochayur commented:
bones system when

actioninja committed:
fix the inverted if clause

actioninja committed:
change to using the emote proc

actioninja committed:
somehow a doubled parenthesis made it in

McBawbaggings commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,65 @@
+		else //you just want to socialize
+			if(bp)
+				playsound(M, get_sfx("desceration"), 50, TRUE, -1)
+				M.visible_message("<span class='warning'>[M] rattles loudly and flails around!!</span>", "<span class='danger'>Your bones so much your missing muscles spasm!!</span>")
```
`Your bones so much` missing a verb here.

McBawbaggings commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -884,3 +884,65 @@
+				M.say("OOF!!", forced="bonehurtingjuice")
+				bp.receive_damage(200, 0, 0) //But I don't think we should
+			else
+				to_chat(M, "<span class='warning'>Your missing arm aches from whever you left it.</span>")
```
`whever` to `wherever`

actioninja commented:
oh dammit it just got merged as well.
